### ISSUE TITLE
Preview cards should not render Warnings or other interactive UIs that are not a preview

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#68131]
### Description

Reported by @t-hamano on https://github.com/WordPress/gutenberg/pull/67433#issuecomment-2553175068

The editor uses 'preview cards' in a few places. For example the list of Patterns 9In the inserter and other places) shows a 'card' with the pattern preview. 

These previews are the actual rendering of the patterns. As such, they also render warnings and potentially other UIs that aren't really part of the preview.

For example, when a Pattern contains a block with invalid markup, it makes perfectly sense to show the Warning in the editing view. Screenshot:

![Image](https://github.com/user-attachments/assets/86fa23e8-d4c0-4b57-8387-031d5eaa31d0)

However, it doesn't make any sense showing the Warning in the card preview. Screenshot:

![Image](https://github.com/user-attachments/assets/a18693f1-7997-4a5a-a6ac-e60e0670ab36)

Such a warning rendered there is hardly usable and I guess it also breaks the accessibility and semantics of the preview card.

### Step-by-step reproduction instructions

- Go to the Site editor > Patterns > and edit one of your custom patterns under 'My patterns'.
- Switch the editor to the 'code' view and alter the markup to intentionally make one of the pattern's block invalid.
- Switch the editor back to the 'visual' view.
- Observe the editor renders the `Block contains unexpected or invalid content.` warning.
- Click 'Open naavigation' at the toop left of the screen and go to 'My patterns'.
- Observe the pattern card preview renders the Warning in the preview.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by karmatosed** [comment]
I agree that showing the warnings in the preview is an issue but my concern is going to be how it looks without the warning. Does it look broken without knowing a reason for example? That said, the warning looks unexpected so I do agree and for now will remove the feedback label to have a suggestion to remove in PR.

**Comment by yogeshbhutkar** [comment]
If we choose not to display the warnings, the interface will appear similar to how it looks within the pattern editor for that specific pattern, except the background color applied by the `has-warning` class will be omitted, and the warning itself will be hidden.

|Before|After|
|-|-|
|![Image](https://github.com/user-attachments/assets/ae0c68cb-838b-4c42-b0e4-3c43f5a68ed8)|![Image](https://github.com/user-attachments/assets/5a10cc2a-06dc-48e5-af2a-2da17bc5ab68)|

However, the warning will be displayed as usual within the editor page for that pattern.
